neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -37,14 +39,596 @@ def save(model, output_dir="./saved_results"):
+            for q_config_key, q_config_value in self.quantization_config.items():
+                if re.search(pattern, q_config_key):
+                    if isinstance(q_config_value, dict) and [algo for algo in q_config_value.keys()][0] == "autoround":
+                        _is_autoround = True
+                    module_quantization_config = [config for config in q_config_value.values()][0]
+
+            if isinstance(module, torch.nn.Linear):
+                # module without qweight means it is not quantized, then skip it
+                loaded_state_dict_keys_set = set(self.loaded_state_dict_keys)
+                if (
+                    name + ".qweight" not in loaded_state_dict_keys_set
+                    and name + ".linear.qweight" not in loaded_state_dict_keys_set
+                ):
+                    continue
+
+                # insert MulLinear module
+                if name + ".linear.qweight" in loaded_state_dict_keys_set:
+                    new_module = MulLinear(module)
+                    set_module(self.original_model, name, new_module)
+                    name += ".linear"
+
+                # replace `torch.nn.Linear` with `WeightOnlyLinear`
+                zp = True if name + ".qzeros" in loaded_state_dict_keys_set else False
+                g_idx = True if name + ".g_idx" in loaded_state_dict_keys_set else False
+
+                kwargs = {}
+                if _is_autoround:
+                    from auto_round.export.export_to_itrex.model_wrapper import (
+                        WeightOnlyLinear as AutoRoundWeightOnlyLinear,
+                    )
+
+                    from .utility import convert_dtype_str2torch
+
+                    WeightOnlyLinearClass = AutoRoundWeightOnlyLinear
+                    kwargs["groupsize"] = module_quantization_config.get("group_size", 32)
+                    kwargs["scale_dtype"] = convert_dtype_str2torch(
+                        module_quantization_config.get("scale_dtype", "fp16")
+                    )
+                else:
+                    from .modules import WeightOnlyLinear as INCWeightOnlyLinear
+
+                    WeightOnlyLinearClass = INCWeightOnlyLinear
+                    kwargs["group_size"] = module_quantization_config.get("group_size", 32)
+                    kwargs["g_idx"] = g_idx
+
+                new_module = WeightOnlyLinearClass(
+                    module.in_features,
+                    module.out_features,
+                    dtype=module_quantization_config.get("dtype", "int"),
+                    bits=module_quantization_config.get("bits", 4),
+                    zp=zp,
+                    bias=module.bias is not None,
+                    use_optimum_format=True,
+                    **kwargs,
+                )
+                set_module(self.original_model, name, new_module)
+        woq_model = self.original_model
+        return woq_model
+
+    def _check_required_packages(self):
```

Comment:
It should be a static method or util function.

Reply:
update in https://github.com/intel/neural-compressor/commit/42cfe39722bad7366a66ff3701e7d3e68e0e31e8: use is_package_available to check package is available or not.